File: loaders/Abstruct Downloader.py
print("Abstruct Downloader")
print("")
print("Importing modules...")
import os
import requests
import urllib
print("Creating directory...")
os.mkdir("Abstruct")
os.chdir("Abstruct")
print("")
print("Loading categories...")
a = requests.get("http://api.abstruct.co/api/packs").json()["data"]
print(str(len(a)) + " categories found.")
for b in a:
	print("")
	print("Creating " + b["name"] + " directory...")
	os.mkdir(b["name"])
	os.chdir(b["name"])
	print("Loading amount of pages...")
	c = requests.get("http://api.abstruct.co/api/packs/" + str(b["id"]) + "/wallpapers").json()["last_page"]
	print(str(c) + " pages found.")
	for d in range(c):
		print("")
		print("Loading page " + str(d + 1) + "...")
		e = requests.get("http://api.abstruct.co/api/packs/" + str(b["id"]) + "/wallpapers?page=" + str(d + 1)).json()["data"]
		print(str(len(e)) + " images found.")
		print("")
		for f in range(len(e)):
			print("Downloading " + e[f]["name"] + "...")
			urllib.request.urlretrieve(e[f]["url_res5"] or e[f]["url_res4"] or e[f]["url_res3"] or e[f]["url_res2"] or e[f]["url_res1"], e[f]["name"] + os.path.splitext(urllib.parse.urlparse(e[f]["url_res5"] or e[f]["url_res4"] or e[f]["url_res3"] or e[f]["url_res2"] or e[f]["url_res1"]).path)[1])
	print("Leaving directory...")
	os.chdir("..")
print("Finished downloading.")

File: loaders/Abstruct Downloader.js
// Show header
console.log('Abstruct Downloader')
console.log('')

// Create directory
require('../tools/directory')('Abstruct')

// Loading categories
console.log('Loading categories...')
const response = require('../tools/request')('http://api.abstruct.co/api/packs', true)
require('fs').writeFileSync('packs', JSON.stringify(response))
const categories = response.data

// Do for each category
categories.forEach(element => {
  console.log('')
  console.log('Downloading category ' + element.name + '...')
  console.log('')
  require('../tools/directory')(element.name)

  // Get amount of pages
  const pages = require('../tools/request')('http://api.abstruct.co/api/packs/' + element.id + '/wallpapers', true).last_page

  // Do for each page
  for (let i = 1; i <= pages; i++) {
    // Get list of wallpapers
    const response = require('../tools/request')('http://api.abstruct.co/api/packs/' + element.id + '/wallpapers?page=' + i, true)
    require('fs').writeFileSync('wallpapers_page=' + i, JSON.stringify(response))
    const data = response.data

    // Download wallpapers
    data.forEach(image => {
      console.log('Downloading image ' + image.name + '...')
      const download = image.url_res5 || image.url_res4 || image.url_res3 || image.url_res2 || image.url_res1 || image.url_thumb
      require('../tools/download')(download, image.name + require('../tools/extension')(download))
    })
  }

  process.chdir('..')
})
